esrally/mechanic/launcher.py

```diff
@@ -352,7 +349,7 @@ def _set_env(self, env, k, v, separator=' '):
             if k not in env:
                 env[k] = v
             else:  # merge
-                env[k] = env[k] + separator + v
+                env[k] = v + separator + env[k]
```

Comment:
I tested this with

```
esrally --test-mode --distribution-version=2.4.6 --on-error=abort --challenge=append-no-conflicts-index-only --telemetry=jfr,gc
```

and indeed you have to preserve the order of the JVM arguments otherwise you'll get an error that the JVM process could not be created. Can you please:

* Change it to your previous implementation so that the order is preserved
* Add a test in ProcessLauncher that adds a telemetry device and checks `ES_JAVA_OPTS`?